internal/clients/stubs/gcp_stub.go

```diff
@@ -60,19 +79,23 @@ func (mock *GCPClientStub) Status(ctx context.Context) error {
 }
 
 func (mock *GCPClientStub) GetInstanceDescriptionByID(ctx context.Context, id, zone string) (*clients.InstanceDescription, error) {
-	return nil, nil
+	instanceDesc := &clients.InstanceDescription{ID: id, PublicIPv4: fmt.Sprintf("10.0.0.%d", counter)}
+	counter = counter + 1
+	return instanceDesc, nil
 }
 
 func (mock *GCPClientStub) ListLaunchTemplates(ctx context.Context) ([]*clients.LaunchTemplate, error) {
 	return nil, nil
 }
 
 func (mock *GCPClientStub) InsertInstances(ctx context.Context, params *clients.GCPInstanceParams, amount int64) ([]*string, *string, error) {
-	return nil, nil, NotImplementedErr
+	ids, err := mock.ListInstancesIDsByLabel(ctx, "1")
+	mock.Instances = ids
```

Comment:
Shall we rather create the ids here in insert method and in List method just return those?

It seems to me, we are completely ignoring the count here, which does not sound like a real implementation.. If I'd change the amount of instances to create in test to 3, it would fail, right?

Reply:
Thank you for the feedback I have disregarded the amount, now I understand what didn't make sense to me 😞